tests/FSharp.Compiler.ComponentTests/Conformance/Types/RecordTypes/AnonymousRecords.fs

```diff
@@ -63,6 +63,28 @@ let x () : {| A: int  |} =  {| A = 123 ; B = ""; C = 1 |}
         |> withDiagnostics [
             (Error 1, Line 2, Col 29, Line 2, Col 58, "This anonymous record has extra fields. Remove fields 'B, C'.")
         ]
+        
+    [<Fact>]
+    let ``Using the wrong anon record with single field`` () =
+        Fsx """
+let x() = ({| b = 2 |} = {| a = 2 |} )
+"""
+        |> compile
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 1, Line 2, Col 26, Line 2, Col 37, "This anonymous record should have field 'b'.")
+        ]
+        
+    [<Fact>]
+    let ``Using the wrong anon record with multiple fields`` () =
+        Fsx """
+let x() = ({| b = 2; c = 3 |} = {| a = 2 |} )
+"""
+        |> compile
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 1, Line 2, Col 33, Line 2, Col 44, "This anonymous record should have fields 'b, c'.")
```

Comment:
Now that I'm looking at it again, we should probably use `'b', 'c'` instead of `'b, c'`.

Reply:
Happy to make the change(It might be a bit tricky). @psfinaki any objections?

Reply:
Ok I will make this happen then :)

Reply:
Updated. Please have a look :)